### PR TITLE
Casper krs/product attributes

### PR DESCRIPTION
**Pull request by casperlotter**
Implementation:

Screenshot 2023-07-25 125426

Zeplin:

Screenshot 2023-07-21 141528
Screenshot 2023-07-21 141510

### CODE REVIEW

File: src/components/Icons/IconCustomer.svelte [hunk 1]
```diff
@@ -1,5 +1,5 @@
 <script lang="ts">
-  export let size: string;
+  export let size: string = '20';
```
Comment (justinengle-krs) [inline src/components/Icons/IconCustomer.svelte:2]: How does this work? The general use case of this size prop is through the <Icon /> element, which passes down a tailwind width. Example:

`<Icon size="smaller" IconComponent={IconCustomer} />`

**Review by justinengle-krs — CHANGES_REQUESTED** [review]

File: src/components/Icons/IconUnlock.svelte [hunk 2]
```diff
@@ -0,0 +1,8 @@
+<svg width="12" height="12" viewBox="0 0 12 12" fill="none" xmlns="http://www.w3.org/2000/svg">
```
Comment (matthampton-stock2shop) [inline src/components/Icons/IconUnlock.svelte:1]: (Maybe not related to this pull request - but noting it)

When trying to manage a template, all of the types are greyed out for existing template attributes:

image

If I add a new one I can choose the type.

File: src/components/Icons/IconUnlock.svelte [hunk 3]
```diff
@@ -0,0 +1,8 @@
...
+  <path
```
Comment (matthampton-stock2shop) [inline src/components/Icons/IconUnlock.svelte:2]: The new screen does not seem to consider the type of the attributes from a template:

image

This one is supposed to have some number ones and some drop-down:

```
{
    "system_meta_template": [
        {
            "id": 700,
            "client_id": 21,
            "created": "2018-06-21 07:46:53",
            "modified": "2023-01-23 11:56:08",
            "name": "product_Laptop",
            "key": "Promotion",
            "default": "",
            "description": "Promotion Status",
            "type": "dropdown",
            "required": 0,
            "values": [
                {
                    "id": "2",
                    "client_id": "21",
                    "created": "2018-06-21 07:47:35",
                    "modified": "2018-06-21 07:47:35",
                    "metatemplate_id": "700",
                    "key": "End of Range",
                    "value": "End of Range"
                },
                {
                    "id": "1",
                    "client_id": "21",
                    "created": "2018-06-21 07:47:12",
                    "modified": "2018-06-21 07:47:35",
                    "metatemplate_id": "700",
                    "key": "Winter Special",
                    "value": "Winter Special"
                }
            ]
        },
        {
            "id": 701,
            "client_id": 21,
            "created": "2018-06-21 07:48:49",
            "modified": "2023-01-23 11:56:08",
            "name": "product_Laptop",
            "key": "CPU",
            "default": "",
            "description": "Processing Power",
            "type": "dropdown",
            "required": 0,
            "values": [
                {
                    "id": "3",
                    "client_id": "21",
                    "created": "2018-06-21 07:49:10",
                    "modified": "2018-06-21 07:49:17",
                    "metatemplate_id": "701",
                    "key": "2.5ghz",
                    "value": "2.5ghz"
                },
                {
                    "id": "4",
                    "client_id": "21",
                    "created": "2018-06-21 07:49:17",
                    "modified": "2018-06-21 07:49:17",
                    "metatemplate_id": "701",
                    "key": "2.7ghz",
                    "value": "2.7ghz"
                }
            ]
        },
        {
            "id": 702,
            "client_id": 21,
            "created": "2018-06-21 07:51:29",
            "modified": "2023-01-23 11:56:08",
            "name": "product_Laptop",
            "key": "Num Pad",
            "default": "false",
            "description": "Number pad on right?",
            "type": "boolean",
            "required": 0,
            "values": []
        },
        {
            "id": 1314,
            "client_id": 21,
            "created": "2018-07-31 15:00:07",
            "modified": "2023-01-23 11:56:08",
            "name": "product_Laptop",
            "key": "Screen Size",
            "default": "",
            "description": "",
            "type": "dropdown",
            "required": 0,
            "values": [
                {
                    "id": "5812",
                    "client_id": "21",
                    "created": "2018-07-31 15:00:15",
                    "modified": "2018-07-31 15:00:19",
                    "metatemplate_id": "1314",
                    "key": "15 inch",
                    "value": "15 inch"
                },
                {
                    "id": "5813",
                    "client_id": "21",
                    "created": "2018-07-31 15:00:19",
                    "modified": "2018-07-31 15:00:19",
                    "metatemplate_id": "1314",
                    "key": "17 inch",
                    "value": "17 inch"
                }
            ]
        },
        {
            "id": 1361,
            "client_id": 21,
            "created": "2018-08-06 11:43:45",
            "modified": "2023-01-23 11:56:08",
            "name": "product_Laptop",
            "key": "Memory",
            "default": "",
            "description": "",
            "type": "dropdown",
            "required": 1,
            "values": [
                {
                    "id": "6225",
                    "client_id": "21",
                    "created": "2018-08-06 11:43:54",
                    "modified": "2019-04-24 12:16:15",
                    "metatemplate_id": "1361",
                    "key": "2gb",
                    "value": "2gb"
                },
                {
                    "id": "6226",
                    "client_id": "21",
                    "created": "2018-08-06 11:43:59",
                    "modified": "2019-04-24 12:16:15",
                    "metatemplate_id": "1361",
                    "key": "4gb",
                    "value": "4gb"
                }
            ]
        }
    ]
}
```

File: src/components/Icons/IconUnlock.svelte [hunk 4]
```diff
@@ -0,0 +1,8 @@
...
+    fill-rule="evenodd"
+    clip-rule="evenodd"
```
Comment (matthampton-stock2shop) [inline src/components/Icons/IconUnlock.svelte:4]: The save changes button gets squished if the title is long:

image

File: src/components/Icons/IconUnlock.svelte (hunk 3 above)
Comment (matthampton-stock2shop) [reply src/components/Icons/IconUnlock.svelte:2]: Is this meant to work?

**Review by matthampton-stock2shop — CHANGES_REQUESTED** [review]

File: src/components/Icons/IconCustomer.svelte [hunk 5]
```diff
@@ -1,5 +1,5 @@
 <script lang="ts">
-  export let size: string;
+  export let size: string = 'smaller';
```
[the remark this answers is not in the document]
Comment (casperlotter) [reply src/components/Icons/IconCustomer.svelte:2]: @justinengle-krs ProductMaster renders the slot in an overflow-y-auto wrapper. Let me know if you have any suggestions for the best way forward.

File: src/components/Icons/IconUnlock.svelte (hunk 4 above)
Comment (casperlotter) [reply src/components/Icons/IconUnlock.svelte:4]: @matthampton-stock2shop This is a PageHeader component styling issue that is not related to this PR. Can you provide some more info around how you produced this? See my attached screenshots for how it is rendering for me.

Screenshot 2023-08-14 093958


Screenshot 2023-08-14 094014

Comment (matthampton-stock2shop) [reply src/components/Icons/IconUnlock.svelte:4]: Logged new issue

**Review by justinengle-krs — APPROVED** [review]

File: src/components/Icons/IconUnlock.svelte (hunk 2 above)
Comment (matthampton-stock2shop) [reply src/components/Icons/IconUnlock.svelte:1]: #208

**Review by matthampton-stock2shop — APPROVED** [review]

**Pull request merged**